### ISSUE TITLE
[Enhancement] Improve trash string search

### ISSUE THREAD

**Issue by igor-ruivo** [DeathbyToast/PvP_IVs#126]
I believe you can catch (potentially) more trash pokémon (that are currently false negatives) with a slight adjustment in the trash string search.
As we commented before, on discord, it's centered around the spread fill step on the code, which has the TODO comment.

For starters, you could delete that:
before:
```
/* TODO: Why is this All League only?? */
/* get actual maxCP and maxHP using spread syntax */
if ((document.getElementById('league').value === "All") && (document.getElementById('trash').checked)) {
    maxCP.fill(Math.max(...maxCP));
    maxHP.fill(Math.max(...maxHP));
}
```

after:
`// do nothing`

The intention is good, but it was just being applied in the context of "All Leagues" which is wrong imo.

Instead, what you could do **only when using trash strings** is, during the `outputCPsHPsBuckets ` step, immediately after using get_matching_string() for the CPs and HPs, append an open range search component for each of them starting at the respective bucket's maxCP/HP + 1, so that you can potentially delete more trash pokémon that follow in that higher range of CP/HP:

before:
```
output.value += "," + get_matching_string(cps[i], lang[1]);
          
if (!trash) {
    output.value += '&!'+i+'*';
}
          
/* skip empty hps */
if (hps[i].size > 0) {
    hps[i] = Array.from(hps[i]);
    hps[i].sort((a, b) => a - b);
    output.value += ',' + get_matching_string(hps[i], lang[2]);
}
```

after:
```
output.value += "," + get_matching_string(cps[i], lang[1]);
          
if (!trash) {
    output.value += '&!'+i+'*';
} else {
    output.value += ',' + lang[1] + (maxCP[i] + 1) + "-";
}
          
/* skip empty hps */
if (hps[i].size > 0) {
    hps[i] = Array.from(hps[i]);
    hps[i].sort((a, b) => a - b);
    output.value += ',' + get_matching_string(hps[i], lang[2]);
    if (trash) {
        output.value += ',' + lang[2] + (maxHP[i] + 1) + "-";
    }
}
```

In summary:
Your initial idea of performing the legacy spread operator on the maxCP/HP was indeed a good idea in order to try to potentially remove more trash pokémon (previously missed/false negatives). I think you tried to discard as much non-targeted-cps/hps as possible, regardless of the respective star, but we can do better by respecting the original star-based max limits and then force-target the remaining bigger values with an open range selector, that depends on the respective star, potentially catching more trash pokémon. Not only that, you can also potentially save a few characters in the string, since the open range selector may comprehend some irrelevant cps/hps combinations previously printed (at least in the All Leagues scenario).

**Comment by DeathbyToast** [comment]
Thanks this is incredibly helpful feedback! Went ahead and got this implemented
